include/camp/tuple.hpp

```diff
@@ -124,23 +124,41 @@ namespace internal
   template <typename... Types, camp::idx_t... Indices>
   struct tuple_helper<camp::idx_seq<Indices...>, camp::list<Types...>>
       : public internal::tuple_storage<Indices, Types>... {
-    template <bool B = concepts::metalib::all_of<
-                  std::is_default_constructible<Types>::value...>::value,
-              typename std::enable_if<B, void>::type* = nullptr>
+
+#if (!defined(__NVCC__)) \
+        || (__CUDACC_VER_MAJOR__ > 10 \
+          || (__CUDACC_VER_MAJOR == 10 && __CUDACC_VER_MINOR__ >= 1))
+    constexpr tuple_helper() = default;
+    constexpr tuple_helper(tuple_helper const&) = default;
+    constexpr tuple_helper(tuple_helper&&) = default;
+    tuple_helper& operator=(const tuple_helper& rhs) = default;
+#else
+    // NOTE: this is to work around nvcc 9 series issues with incorrect
+    // creation of defaulted constructors
+    template <
+        bool B = concepts::metalib::all_of<std::is_default_constructible<Types>::value...>::value,
+        typename std::enable_if<B, void>::type* = nullptr>
     CAMP_HOST_DEVICE constexpr tuple_helper()
     {
     }
     CAMP_HOST_DEVICE constexpr tuple_helper(tuple_helper const& rhs)
         : tuple_storage<Indices, Types>(
-              rhs.tuple_storage<Indices, Types>::get_inner())...
+            rhs.tuple_storage<Indices, Types>::val)...
     {
     }
     CAMP_HOST_DEVICE constexpr tuple_helper(tuple_helper&& rhs)
         : tuple_storage<Indices, Types>(
-              std::forward<Types>(rhs.tuple_storage<Indices, Types>::val))...
+            std::forward<Types>(rhs.tuple_storage<Indices, Types>::val))...
     {
     }
 
+    CAMP_HOST_DEVICE tuple_helper& operator=(const tuple_helper& rhs)
```

Comment:
@trws @rhornung67 If I change this to `tuple_helper& operator=(const tuple_helper& rhs) = default;` (without the body of course), then CUDA9 is happy. This is baffling though because a templated version of this copy assignment constructor is defined on line 180.

Reply:
Ok, I'll push a patch in a minute that should resolve this then.  We do need to have both, because some things can't compile without them, but why the explicit one is dying in raja I'm not sure.  If 9 is working though I should be able to just hoist the defaulted one out of the check.

Reply:
@trws Looks good! Shall we merge this? If so, should we use this version of CAMP for the imminent RAJA release, or are there other CAMP PRs we should look at?

Reply:
If this works for you I'll merge, and we should probably use this, there is other stuff I'd really like to get in but that's pending the C++14 feedback.

Reply:
Yes, it works for RAJA with CUDA 9 and 10, so we should merge this change. @rhornung67 can then proceed with the RAJA RC branch. I don't know about the C++14 usage status.